### PR TITLE
Add SourceSpanWithContextExtension.subspan

### PR DESCRIPTION
**Pull request by nex3**

### CODE REVIEW

File: lib/src/utils.dart [hunk 1]
```diff
@@ -2,7 +2,12 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:charcode/charcode.dart';
```
Comment (natebosch) [inline lib/src/utils.dart:5]: Import the local `charcode.dart` instead.

File: lib/src/utils.dart [hunk 2]
```diff
@@ -2,7 +2,12 @@
...
+import 'package:tuple/tuple.dart';
```
Comment (natebosch) [inline lib/src/utils.dart:6]: We want to avoid new dependencies on this package. How about writing a `SubSpan` class?

**Review by natebosch — CHANGES_REQUESTED** [review]

File: lib/src/utils.dart (hunk 1 above)
Comment (nex3) [reply lib/src/utils.dart]: Done.

File: lib/src/utils.dart (hunk 2 above)
Comment (nex3) [reply lib/src/utils.dart:5]: A "subspan" isn't really identified as a pair of locations... this is just a multiple return. Is there really no general-purpose way of doing that anymore?

File: pubspec.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ dependencies:
   collection: ^1.15.0
   path: ^1.8.0
   term_glyph: ^1.2.0
+  tuple: ^2.0.0
```
Comment (kevmoo) [inline pubspec.yaml:14]: Do we need this dependency? Could we just copy-paste?

This package is used ALL OVER. We'd have to bring the dep back into the Dart SDK, etc

File: lib/src/utils.dart (hunk 2 above)
Comment (natebosch) [reply lib/src/utils.dart:5]: We don't currently have a general purpose way of retuning multiple values in our core packages or any packages in the SDK. We are using small single use classes for now, and will hopefully eventually replace them with language supported tuples.

File: lib/src/utils.dart [hunk 4]
```diff
@@ -89,3 +92,53 @@ int? findLineStart(String context, String text, int column) {
   // ignore: avoid_returning_null
   return null;
 }
+
+/// Returns the start and end locations of the span from [start] code units
```
Comment (kevmoo) [inline lib/src/utils.dart:96]: nit: maybe explicitly say it's a 2-element List?

**Review by natebosch — APPROVED** [review]

**Pull request merged**